rides/__init__.py

```diff
@@ -381,9 +395,76 @@ def leave_ride(car_id, rider_username):
     username = current_user.id
     car = Car.query.get(car_id)
     rider = Rider.query.filter(Rider.username == rider_username, Rider.car_id == car_id).first()
+    if rider.username == username and rider is not None:
+        db.session.delete(rider)
+        car.current_capacity -= 1
+        notify_opening( car.event_id, car.name, car_id )
+        db.session.add(car)
+        db.session.commit()
+    return redirect(url_for('index'))
+
+# Automatically leave a ride and join different ride, used for Needs a Ride Notifications
+@app.route('/autojoin/<string:leave_id>/<string:join_id>/<string:user>', methods=['GET'])
+@login_required
+def autojoin(leave_id, join_id, user):
+    # LEAVE
+    username = current_user.id
+    car = Car.query.get(leave_id)
+    rider = Rider.query.filter(Rider.username == user, Rider.car_id == leave_id).first()
     if rider.username == username and rider is not None:
         db.session.delete(rider)
         car.current_capacity -= 1
         db.session.add(car)
         db.session.commit()
+    # JOIN
+    #username = current_user.id
+    name = current_user.firstname + " " + current_user.lastname
+    slack = current_user.slack
+    email = current_user.email
+    acc_type = current_user.acc_type
+    car = Car.query.get(join_id)
+    event = Event.query.get(car.event_id)
+    attempted_username = user
+    if attempted_username == username:
+        for c in event.cars:
+            incar = ( c.username == username ) or ( username in c.riders )
```

Comment:
`incar = event.cars[-1].username == username or username in evenv.cars[-1].riders`, which probably isn't what you intended given the for.

This will check if the user is in any of the cars
```suggestion
		incar = any(map(lambda c: c.username == username or username in c.riders, event.cars))
```

alternatively, avoiding `map()` and `any()`:
```suggestion
		incar = False
		for c in event.cars:
			if ( c.username == username ) or ( username in c.riders ):
				incar = True
				break
```

Reply:
Would this work the same?
`
for c in event.cars:
            incar = ( c.username == username ) or ( username in c.riders )
            if incar:
                        break
`